src/darwin/Framework/CHIP/MTRBackwardsCompatShims.h

```diff
@@ -0,0 +1,37 @@
+/**
+ *    Copyright (c) 2023 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#import <Foundation/Foundation.h>
+
+#import <Matter/MTRCommandPayloadsObjc.h>
+
+/**
+ * This file defines manual backwards-compat shims of various sorts to handle
+ * API changes that happened.
+ */
+
+NS_ASSUME_NONNULL_BEGIN
+
+@interface MTRGroupKeyManagementClusterKeySetReadAllIndicesParams ()
+/**
+ * This command used to incorrectly have a groupKeySetIDs field.
+ */
+@property (nonatomic, copy) NSArray * groupKeySetIDs API_AVAILABLE(ios(16.1), macos(13.0), watchos(9.1), tvos(16.1))
```

Comment:
Did you test accessing this property from code? I don't think this will be synthesized unless you arrange for this header to be included in the `.mm` file. The best way to handle it might be to just implement them by hand to return `@[]` for the getter and a no-op setter.

Reply:
Oh, good catch.  That in fact did not work right.

Doing the manual impl seems hard, since I would need to somehow get it in the right place in the file, no? But including the shim header from the right .mm is quite easy, so I will do that.